src/Protocol/PortableContact.php

```diff
@@ -1227,41 +1249,20 @@ public static function checkServer($server_url, $network = "", $force = false)
 		$platform = strip_tags($platform);
 
 		if ($servers) {
-			q(
-				"UPDATE `gserver` SET `url` = '%s', `version` = '%s', `site_name` = '%s', `info` = '%s', `register_policy` = %d, `poco` = '%s', `noscrape` = '%s',
-				`network` = '%s', `platform` = '%s', `last_contact` = '%s', `last_failure` = '%s' WHERE `nurl` = '%s'",
-				dbesc($server_url),
-				dbesc($version),
-				dbesc($site_name),
-				dbesc($info),
-				intval($register_policy),
-				dbesc($poco),
-				dbesc($noscrape),
-				dbesc($network),
-				dbesc($platform),
-				dbesc($last_contact),
-				dbesc($last_failure),
-				dbesc(normalise_link($server_url))
-			);
+			$fields = ['url' => $server_url, 'version' => $version,
+					'site_name' => $site_name, 'info' => $info, 'register_policy' => $register_policy,
+					'poco' => $poco, 'noscrape' => $noscrape, 'network' => $network,
+					'platform' => $platform, 'registered-users' => $registered_users,
+					'last_contact' => $last_contact, 'last_failure' => last_failure];
+			dba::update('gserver', $fields, ['nurl' => normalise_link($server_url)]);
 		} elseif (!$failure) {
-			q(
-				"INSERT INTO `gserver` (`url`, `nurl`, `version`, `site_name`, `info`, `register_policy`, `poco`, `noscrape`, `network`, `platform`, `created`, `last_contact`, `last_failure`)
-						VALUES ('%s', '%s', '%s', '%s', '%s', %d, '%s', '%s', '%s', '%s', '%s', '%s', '%s')",
-				dbesc($server_url),
-				dbesc(normalise_link($server_url)),
-				dbesc($version),
-				dbesc($site_name),
-				dbesc($info),
-				intval($register_policy),
-				dbesc($poco),
-				dbesc($noscrape),
-				dbesc($network),
-				dbesc($platform),
-				dbesc(datetime_convert()),
-				dbesc($last_contact),
-				dbesc($last_failure),
-				dbesc(datetime_convert())
-			);
+			$fields = ['url' => $server_url, 'nurl' => normalise_link($server_url), 'version' => $version,
```

Comment:
What’s the difference between the update fields and the insert fields? Any chance we can have only one fields array?

Reply:
In "insert" we fill the "created" field and the "nurl" field that we don't fill at upfate.